File: src/services/TheBrain/src/domain/geolocation/IGeoLocationCalculator.ts
export default interface IGeoLocationCalculator {
    isInsideRadius(
        centerLatitude : number,
        centerLongitude : number,
        targetLatitude : number,
        targetLongitude : number,
        radius : number
    ) : boolean;

    calculateDistance(
        firstLatitude : number,
        firstLongitude : number,
        secondLatitude : number,
        secondLongitude : number
        ) : number;

    degreesToRadians(degrees : number) : number;
    kilometerRadiusToDegreesRadius(kilometerRadius : number) : number;
}

File: src/services/NationalGeographicMessager/Domain/GeolocationAggregated/IGeoLocationCalculator.cs
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace NationalGeographicMessager.Domain.GeolocationAggregated
{
    public interface IGeoLocationCalculator
    {
        bool IsInsideRadius(
            double centerLatitude,
            double centerLongitude,
            double targetLatitude,
            double targetLongitude,
            double radius
            );

        double KilometerRadiusToDegreesRadius(double kilometerRadius);
    }
}
